generate/generate.go

```diff
@@ -21,7 +21,14 @@ var (
 
 // Generator represents a generator for a container spec.
 type Generator struct {
-	spec *rspec.Spec
+	spec       *rspec.Spec
+	ExportOpts *ExportOptions
```

Comment:
Oh I see what you're saying (hopefully). I think the solution is not to tie this `ExportOption` to the `Generator` but just be a _dangling_ option struct which is directly provided to `Save*`. Does this work for you?